### DISCUSSION TITLE
Taiga UI 5

### DISCUSSION THREAD

**Discussion by nsbarsukov** [taiga-family/taiga-ui#8353] [RFC]
- [x] Switch all dynamic things like formatting, i18n etc from Observables to signals
- [x] Remove usage of `"experimentalDecorators": true` and rewrite all existing decorators using only modern syntax
- [x] Remove `@angular/animations` from peer deps
- [x] Rename left/right to start/end everywhere for better RTL semantics
- [x] https://github.com/taiga-family/taiga-ui/issues/11618#issuecomment-3227246123
- [x] https://github.com/taiga-family/taiga-ui/pull/12785
- [x] https://github.com/taiga-family/taiga-ui/issues/8343 consider this breaking change
- [x] Update Angular to 19.x.x
- [x] Enable built-in support for dark mode in `Root` with a new `WritableSignal<boolean>` token
- [x] [Normalize](https://github.com/taiga-family/taiga-ui/pull/12692) `TuiPoint` as `[x: number, y: number]` across the app and use it everywhere
- [x] Review animations, some need more params, some are no longer used
- [x] Review portals, maybe we can just show them all in one stack for simplicity in order of their creation
- [x] Toss around utils SEP to move all DI related ones into DI SEP inside CDK, move copy to components, move fullscreen to kit, review all other entities placement
- [x] Fix `closeable` typo to `closable`
- [x] Refactor [Stepper](https://taiga-ui.dev/navigation/stepper) to use [Cell](https://taiga-ui.dev/layout/cell) ?
- [x] Move Surface to layout?
- [x] Refactor [Accordion](https://taiga-ui.dev/components/accordion)
- [x] Refactor [Pagination](https://taiga-ui.dev/navigation/pagination).
    We need to break it down to 2 components, dots and number buttons.
- [x] Refactor dialogs
    We need easier way to reuse built-in logic like closeable/dismissible for custom dialogs
- [x] ~Rename `@taiga-ui/icons` to `@taiga-ui/icons-lucide`~
    Instead we need to drop font-awesome and material icon packages due to added icon font support
- [x] [Update breakpoints](https://github.com/taiga-family/taiga-ui/pull/12642) to 0 < mobile < 768 < desktopSmall < 1280 < desktopLarge < ∞
- [x] ~Use tokens for animations (duration + timing-function) from `@taiga-ui/design-tokens/palette/animation.css`~ non breaking, we will do it after release
- [x] [Make `Drawer` in overlay with `top: 0` by default](https://github.com/orgs/taiga-family/discussions/8353)
- [x] Refactor `string | TuiStringHandler<T>` for icons to always use functions, consider changing icon dictionaries to functions too
- [x] Add appearance as host directive to cards
- [x] `tuiIsNativeFocused` => `tuiIsFocused`  & `tuiIsNativeFocusedIn` => `tuiIsFocusedIn `
- [x] Remove css @supports (...) for already supported features (i.e. `inset-inline-end`)

**Comment by KhaledGabr** [comment]
Speaking of Carousel, Maybe you could consider integrating with embla-carousel .. I ended up going with it instead of using Taiga's one for my project. 

https://www.embla-carousel.com/get-started/

**Comment by SkyZeroZx** [comment]
I've been reviewing the library lately and it seems great with all the addons and functionalities like aesthetics.
I'm still looking at the code but I still have a doubt if TaigaUI is completely friendly with Angular's SSR, it could be considered as an improvement since several components are going to be refactored which would make it an even better library than it already is.

**Reply by waterplea** [reply]
In v4 our entire demo portal is prerendered with all the routes — this is how we test that all our components work with SSR, this is done as a pipeline on each PR. If you see a particular issue with SSR — please report it, SSR support is important to us.

**Comment by vytautas-pranskunas-** [comment]
May I ask why Angular 17? if there is already 19 and by the time of V5 release there will be 20 for sure :)

**Reply by waterplea** [reply]
There's no need for us to go above 17 at this moment, all we need is input and query signals. 18 and 19 did not deliver anything crucial in library development and we try to stay on lowest possible version so our userbase can use Taiga UI without requirement to update Angular version. If v5 takes a bit too long we might move to 18 after all. Things in 18 we might want:
- `ng-content` fallback
- form events
- `@let` syntax

**Reply by hakimio** [reply]
One more good reason to go with `v18` is zoneless CD.

**Reply by waterplea** [reply]
> One more good reason to go with `v18` is zoneless CD.

That's for apps, for libs it is at most easier to test with that mode and while it is experimental we should stick to zonefull as our default environment.

**Reply by hakimio** [reply]
It's for libs in the sense that you'll be able to test if the library works well in both `zoneless` and `zonefull` modes. I think there is a lot of interest in zoneless and being able to prove that Taiga works in zoneless would be a big advantage.

**Reply by KhaledGabr** [reply]
I have been using zoneless with taiga for a while, and didn't face any issues.

**Reply by hakimio** [reply]
I have: https://github.com/orgs/taiga-family/discussions/7221#discussioncomment-10623336

**Reply by hakimio** [reply]
As of `Angular v20`, `zoneless` mode is no longer experimental. Would be great if last few Taiga issues would be addressed to be fully `zoneless` compatible.

**Reply by waterplea** [reply]
Accordion is now completely refactored and should work fine and dialogs on mobile are meant to use `SheetDialog` now (for example using `TuiResponsiveDialogService`). Here's a zoneless Angular 20 started with Taiga UI, could you reproduce the issues you still have there so we can fix them?

**Reply by waterplea** [reply]
https://stackblitz.com/edit/taiga-ui-zoneless

**Reply by hakimio** [reply]
The last issue was with dialogs. Can the normal dialog created with `TuiDialogService` from `@taiga-ui/core` now adapt to mobile screen when running in `zoneless` mode (it definitely works in zoneful mode)? I really don't want to migrate to another dialog service everywhere just to solve this issue.
Also, the last time I tried, some of the dialog e2e/playwright tests where failing when running in `zoneless` mode or with zone `event coalescing` enabled. Do the tests pass now in `zoneless` mode? Specifically the `routed dialog` tests.

**Reply by waterplea** [reply]
I've added the fullscreen normal dialog to the starter and I think it works as expected. We do not yet run tests in zoneless since we're still on Angular 16 but this summer we should upgrade and switching to zoneless for tests sounds like a good idea.

**Reply by hakimio** [reply]
Can't reproduce the dialog issue in the demo app anymore. The old issue was that the dialog would not change to mobile mode when window is resized or mobile device emulation is enabled. 
Will test this out again in my application when migrating to Angular 20.
Thanks for looking into the issues 🙂

**Reply by hakimio** [reply]
Everything seems to be working correctly in `zoneless` mode with Angular 20 in my application now too.

**Comment by hakimio** [comment]
@waterplea not a fan of tiny (13px) text and tiny paddings in the new docs. Everything seems to be cramped and uncomfortable there. I'd say that's the opposite of modern UI. Can you at least increase the 13px font to 14px?

**Reply by waterplea** [reply]
Our internal design system uses 13px/15px fonts, but historically we were using 16px in open source theme to avoid iOS zoom on inputs. I guess it's a good idea to bump 13px to 14px as well in the next major release (also bumped padding in navigation a little bit):
https://github.com/taiga-family/taiga-ui/pull/13160

**Reply by hakimio** [reply]
Thanks 🙂
